exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -288,4 +288,51 @@ object SchemaUtils {
             currentDialect.resetCaches()
         }
     }
+
+    /**
+     * Sets the current default schema to [schema]
+     *
+     * @sample org.jetbrains.exposed.sql.tests.shared.SchemaTests
+     */
+    fun setSchema(schema: Schema) {
+        with(TransactionManager.current()) {
+            val createStatements = schema.setSchemaStatement()
+
+            execStatements(false, createStatements)
+
+            if(currentDialect is MysqlDialect) {
```

Comment:
Could you comment on why we need to set a catalog for MySQL and not set a schema for other databases?

Reply:
This is a workaround for mysql jdbc version lower or equal to 5.1.48. 
In this version, the `connection.catalog` doesn't change after executing `USE db` statement. So in this workaround I'm setting it manually.

For other databases, executing the command that changes the schema sets automatically the connection.schema field to the name of the schema we are switching to. So `connection.setSchema` will be useless.

I tried with Mysql jdbc version 6 and higher and it works fine. After  executing `USE db` statement, the `connection.catalog` returns the new schema name  without setting it manually. Also for MariaDB(which supports only catalogs just like mysql) , the problem doesn't appear.